### PR TITLE
#4 Alter post-processing methods to reduce occurrence of repeated code (WIP)

### PR DESCRIPTION
**Pull request by yaseminbridges**
Updated methods for post-processing of Exomiser json results into PhEval gene and variant output using the "common" methods moved to the PhEval repo. This should perform as it should once [PR#54 ](https://github.com/monarch-initiative/pheval/pull/54#issue-1572534718) in the PhEval repo is merged

### CODE REVIEW

**Review by matentzn — APPROVED** [review]
This seems to be mostly a refactoring PR, I don't have anything to add. @souzadevinicius?

File: src/pheval_exomiser/post_process/post_process_results_format.py [hunk 1]
```diff
@@ -114,81 +23,131 @@ def read_exomiser_json_result(exomiser_result_path: Path) -> dict:
...
 
-    def simplify_gene_result(self) -> [dict]:
-        """Simplify Exomiser json output into gene results."""
+    @staticmethod
+    def find_gene_symbol(result_entry: dict) -> str:
+        """Return gene symbol from Exomiser result entry."""
+        return result_entry["geneSymbol"]
+
+    @staticmethod
+    def find_gene_identifier(result_entry: dict) -> str:
```
Comment (julesjacobsen) [inline src/pheval_exomiser/post_process/post_process_results_format.py:37]: Are these private methods? If so _ them.

File: src/pheval_exomiser/post_process/post_process_results_format.py [hunk 2]
```diff
@@ -114,81 +23,131 @@ def read_exomiser_json_result(exomiser_result_path: Path) -> dict:
     return exomiser_result
 
 
-class StandardiseExomiserResult:
-    """Standardise Exomiser output into simplified gene and variant results for analysis."""
-
+class PhEvalGeneResultFromExomiserJsonCreator:
     def __init__(self, exomiser_json_result: [dict], ranking_method: str):
```
Comment (julesjacobsen) [inline src/pheval_exomiser/post_process/post_process_results_format.py:27]: would `ranking_method` be better named `score_name`? e.g. the value toud be `combinedScore` , `phenotypeScore`, or `pValue`

File: src/pheval_exomiser/post_process/post_process_results_format.py [hunk 3]
```diff
@@ -114,81 +23,131 @@ def read_exomiser_json_result(exomiser_result_path: Path) -> dict:
...
+    @staticmethod
+    def find_alt(result_entry: dict) -> str:
+        """Return alternate allele from Exomiser result entry."""
+        return result_entry["alt"]
+
+    def find_relevant_score(self, result_entry) -> float:
+        """Return score from Exomiser result entry."""
+        return round(result_entry[self.ranking_method], 4)
+
+    def extract_pheval_variant_requirements(self) -> [PhEvalVariantResult]:
```
Comment (julesjacobsen) [inline src/pheval_exomiser/post_process/post_process_results_format.py:95]:
```python
@staticmethod
def extract_pheval_variant_requirements(self, exomiser_json_result: [dict], ranking_method: str) -> [PhEvalVariantResult]:
```
Also consider the name? Although this isn't really a public API it's  important so you understand its purpose next month :)

**Review by julesjacobsen — APPROVED** [review]
Looks cool. This will break when the main pheval changes go in, but that's for another PR...

**Pull request merged**